style/infra-style.md

```diff
@@ -0,0 +1,109 @@
+
+
+## Our Goals:
+- improve maintainability
+- reduce cognitive load when reading code; we grok it faster if choices are consistent across codebases
+- avoid mistakes, e.g. typos
+- continually improve code
+- follow best practices in broader community
+- avoid disagreements about style
+
+- do as much as possible with automated tools
+
+## Tools
+
+Ruby:
+  - [rubocop](https://rubocop.org/) [cop documentation](https://docs.rubocop.org/rubocop) [github](https://github.com/rubocop/rubocop)
+  - [rubocop-rspec cop documentation](https://docs.rubocop.org/rubocop-rspec/cops.html) [github](https://github.com/rubocop/rubocop-rspec)
+  - [rubocop-rails cop documentation](https://docs.rubocop.org/rubocop-rails/cops.html) [github](https://github.com/rubocop/rubocop-rails)
+  - (not often used by us)[rubocop-performance cop documentation](https://docs.rubocop.org/rubocop-performance/cops_performance.html) [github](https://github.com/rubocop/rubocop-performance)
+
+We want these tools to be exercised when a pull request is created, at a minimum.
+
+
+## Editing existing codebases
+
+Most of our work is editing existing code bases, whether they are 3 months old or 15 years old.
+
+### RULE:  Follow the existing rules.  Strive for no new exceptions.
+
+### Should it be a goal to reduce .rubocop_todo.yml to empty?
+
+(TBD)
+
+
+## Starting a New Code Base
+
+Start with default style guides + exceptions below.
+
+
+## When Style exceptions are acceptable
+
+For long classes/methods/blocks, consider SOLID:
+
+    **S**ingle responsibility
+    **O**pen-Closed (open for extension, closed for modification)
+    **L**iskov substitution
+    **I**nterface segregation (clients should not be forced to depend of interfaces they don't use)
+    **D**ependency inversion (depend on abstractions, not concretions)
+
+However, refactoring can reduce readability:
+- the reader now has to jump around more in the code without a clear gain in readability
+- method/class names matter!
+
+### When can we turn a rule off?
+
+(TBD)
+
+
+## What to do when keeping an exception
+
+- Prefer an inline exception to one in .rubocop.yml
+   - On same line if it's a single line to be excepted and one rule
+   - Above disable, below disable if multiple rules or multiple lines excepted
+- Consider a comment indicating why ("shameless green", "we need UTF-8 here")
+
+
+## How / When apply new linter rules?
+
+For rubocop, this means adding new rules to the config file.
+
+Guidance:
+- when warnings are annoying
+- when it's more than a handful
+- when working in the code base for whatever reason
+
+
+# Unenforceable Styles
+
+## On Comments:
+- if you read the code out loud and it says what the comment says -- the comment is superfluous
+
+
+## On Names:
+
+Good names make code a lot easier to read.  Choose illustrative names for methods, for variables, for classes.
+
+- Prefer shorter over longer if intent is revealed clearly
```

Comment:
I'm not sure I agree with this bit of guidance. Maybe examples would help? Or maybe we should drop the non-Rubocop stuff here or break it out into a separate conversation to keep the scope on this tight?

Reply:
I added examples.

Reply:
K, the examples help. Thanks!